File: FeedItForward/backend/app/schemas/driver.py
from pydantic import BaseModel

from .user import User, UserCreate, UserUpdate, Role

class Driver(BaseModel):
    driver_id: int
    vehicle_number: str
    licence_number: str

    user_id: int
    user: User

    class Config:
        orm_mode = True

class DriverCreate(UserCreate):
    role: Role = Role.DRIVER
    vehicle_number: str
    licence_number: str

class DriverUpdate(UserUpdate):
    driver_id: int
    vehicle_number: str
    licence_number: str

File: FeedItForward/frontend/src/schemas/driver.ts
import { UserCreate, UserDisplay, UserUpdate } from "./user";

export interface Driver {
  driver_id: number;
  vehicle_number: string;
  licence_number: string;

  user_id: number;
  user: UserDisplay;
}

export interface DriverCreate extends UserCreate {
  vehicle_number: string;
  licence_number: string;
}

export interface DriverUpdate extends UserUpdate {
  driver_id: number;
  vehicle_number: string;
  licence_number: string;
}
